crates/polars-plan/src/dsl/function_expr/temporal.rs

```diff
@@ -78,6 +81,29 @@ pub(super) fn datetime(
     use polars_core::export::chrono::NaiveDate;
     use polars_core::utils::CustomIterTools;
 
+    if s.iter().any(|s| s.is_empty()) {
+        return Ok(Column::new_empty(
+            PlSmallStr::from_static("datetime"),
+            &DataType::Datetime(
+                time_unit.to_owned(),
+                match time_zone {
+                    #[cfg(feature = "timezones")]
+                    Some(time_zone) => {
+                        validate_time_zone(time_zone)?;
+                        Some(PlSmallStr::from_str(time_zone))
+                    },
+                    _ => {
+                        polars_ensure!(
+                            time_zone.is_none(),
```

Comment:
This should be an assert. We panic on missing features.

Reply:
Thanks for the info! Updated in 8b485625e3bbdf063e4e88ca27156ff0ee3c4129. The existing `polars_ensure` on line 200 has also been replaced with `assert`.